pkg/config/config_template.yaml

```diff
@@ -4058,6 +4058,80 @@ api_key:
+# reverse_dns_enrichment:
+
+  ## @param workers - integer - optional - default: 10
+  ## The number of concurrent workers used perform reverse DNS lookups.
+  # workers: 10
+
+  ## @param chan_size - integer - optional - default: 5000
+  ## The size of the channel used to send reverse DNS lookups requests to the workers.
+  # chan_size: 5000
+
+  ## @param cache - custom object - optional
+  ## This section configures the cache used by the reverse DNS enrichment component.
+  # cache:
+
+    ## @param enabled - boolean - optional - default: true
+    ## Set to true to enable reverse DNS enrichment caching.
+    #
+    # enabled: true
+
+    ## @param entry_ttl - duration - optional - default: 24h
+    ## The amount of time that a cache entry remains valid before it is expired and removed from the cache.
+    # entry_ttl: 24h
+
+    ## @param clean_interval - duration - optional - default: 2h
+    ## An interval that specifies how often expired entries are removed from the cache to free space.
+    # clean_interval: 2h
+
+    ## @param persist_interval - duration - optional - default: 2h
+    ## An interval that specifies how often the cache is persisted to disk so the cache can be reloaded when the agent is upgraded or restarted.
+    # persist_interval: 2h
+
+    ## @param max_retries - integer - optional - default: 10
+    ## The maximum number of retries to perform when a DNS lookup operation fails, after which the hostname "" is returned and cached for the IP address.
+    # max_retries: 10
+
+    ## @param max_size - integer - optional - default: 1000000
+    ## The maximum size in entries of the cache, above which additional entries will not be cached.
+    #
+    # max_size: 1000000
+
+  ## @param rate_limiter - custom object - optional
+  ## This section configures the rate limiter used by the reverse DNS enrichment component.
+  # rate_limiter:
+
+    ## @param enabled - boolean - optional - default: true
+    ## Set to true to enable the reverse DNS enrichment rate limiter.
+    #
+    # enabled: true
+
+    ## @param limit_per_sec - integer - optional - default: 1000
+    ## The maximum number of reverse DNS lookups allowed per second by the rate limiter.
+    # limit_per_sec: 1000
+
+    ## @param limit_throttled_per_sec - integer - optional - default: 1
+    ## The maximum number of reverse DNS lookups allowed per second when the rate limiter is throttled due to errors exceeding the threshold.
+    # limit_throttled_per_sec: 1
+
+    ## @param throttle_error_threshold - integer - optional - default: 10
+    ## The number of consecutive errors that will trigger the rate limiter to throttle down to limit_throttled_per_sec.
+    # throttle_error_threshold: 10
+
+    ## @param recovery_intervals - integer - optional - default: 5
+    ## The number of intervals over which to throttle the rate limit back to limit_per_sec after being throttled.
```

Comment:
small suggestion?: i don't know if there's a limit to how long the description would be - but maybe it would be worth giving an example with what this would look like for additional clarity? (or maybe documentation folks lmk if this already makes sense them!)

i.e. 
every 5 seconds (recovery interval), slowly ramp up over 5 (recovery intervals) back to 1000 (limit per seconds)

not sure if the anyone has a better way to put this than me, but just thought it's hard to grasp initially with so many configs and the name `recovery_interval` close to this one 😅

Reply:
Added more details and an example 32b7ba27c26032b39ad63758063e84f18fc1d558